<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsValueRelationConfigDlgBase</class>
 <widget class="QWidget" name="QgsValueRelationConfigDlgBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>342</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="0,0">
   <item row="2" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Key column</string>
     </property>
     <property name="buddy">
      <cstring>mKeyColumn</cstring>
     </property>
    </widget>
   </item>
   <item row="14" column="0">
    <widget class="QCheckBox" name="mUseCompleter">
     <property name="text">
      <string>Use completer</string>
     </property>
    </widget>
   </item>
   <item row="10" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Description</string>
     </property>
    </widget>
   </item>
   <item row="16" column="1">
    <widget class="QgsSpinBox" name="mNofColumns">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>Group column</string>
     </property>
     <property name="buddy">
      <cstring>mGroupColumn</cstring>
     </property>
    </widget>
   </item>
   <item row="20" column="0" colspan="2">
    <widget class="QTextEdit" name="mFilterExpression"/>
   </item>
   <item row="1" column="1">
    <widget class="QgsMapLayerComboBox" name="mLayerName"/>
   </item>
   <item row="2" column="1">
    <widget class="QgsFieldComboBox" name="mKeyColumn"/>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>Select layer, key column and value column</string>
     </property>
    </widget>
   </item>
   <item row="15" column="0" colspan="2">
    <widget class="QCheckBox" name="mAllowMulti">
     <property name="text">
      <string>Allow multiple selections</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Layer</string>
     </property>
     <property name="buddy">
      <cstring>mLayerName</cstring>
     </property>
    </widget>
   </item>
   <item row="11" column="0" colspan="2">
    <widget class="QCheckBox" name="mAllowNull">
     <property name="text">
      <string>Allow NULL value</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>Value column</string>
     </property>
     <property name="buddy">
      <cstring>mValueColumn</cstring>
     </property>
    </widget>
   </item>
   <item row="14" column="1">
    <widget class="QCheckBox" name="mCompleterMatchFromStart">
     <property name="text">
      <string>Only match from the beginning of the string</string>
     </property>
    </widget>
   </item>
   <item row="17" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="label_19">
       <property name="text">
        <string>Filter expression</string>
       </property>
       <property name="buddy">
        <cstring>mValueColumn</cstring>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="mEditExpression">
       <property name="layoutDirection">
        <enum>Qt::RightToLeft</enum>
       </property>
       <property name="icon">
        <iconset resource="../../../images/images.qrc">
         <normaloff>:/images/themes/default/mIconExpression.svg</normaloff>:/images/themes/default/mIconExpression.svg</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="8" column="1">
    <widget class="QgsFieldComboBox" name="mGroupColumn"/>
   </item>
   <item row="9" column="0" colspan="2">
    <widget class="QCheckBox" name="mDisplayGroupName">
     <property name="text">
      <string>Display group name</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QgsFieldComboBox" name="mValueColumn"/>
   </item>
   <item row="16" column="0">
    <widget class="QLabel" name="label_nofColumns">
     <property name="text">
      <string>Number of columns</string>
     </property>
    </widget>
   </item>
   <item row="10" column="1">
    <widget class="QgsFieldExpressionWidget" name="mDescriptionExpression" native="true"/>
   </item>
   <item row="4" column="0" rowspan="2" colspan="2">
    <widget class="QgsCollapsibleGroupBox" name="mOrderByGroupBox">
     <property name="title">
      <string>Order By</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="mOrderByKey">
        <property name="toolTip">
         <string>Use the values in the key column for sorting</string>
        </property>
        <property name="text">
         <string>Key</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="mOrderByValue">
        <property name="toolTip">
         <string>Use the values in the value column for sorting</string>
        </property>
        <property name="text">
         <string>Value</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QRadioButton" name="mOrderByField">
          <property name="toolTip">
           <string>Use the values from a specific field for sorting</string>
          </property>
          <property name="text">
           <string>Field</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QgsFieldComboBox" name="mOrderByFieldName"/>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="mOrderByDescending">
        <property name="toolTip">
         <string>Reverse the sorting</string>
        </property>
        <property name="text">
         <string>Descending order</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsSpinBox</class>
   <extends>QSpinBox</extends>
   <header>qgsspinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsfieldcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldExpressionWidget</class>
   <extends>QWidget</extends>
   <header>qgsfieldexpressionwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>qgscollapsiblegroupbox.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>mLayerName</tabstop>
  <tabstop>mKeyColumn</tabstop>
  <tabstop>mValueColumn</tabstop>
  <tabstop>mGroupColumn</tabstop>
  <tabstop>mDescriptionExpression</tabstop>
  <tabstop>mAllowNull</tabstop>
  <tabstop>mAllowMulti</tabstop>
  <tabstop>mUseCompleter</tabstop>
  <tabstop>mCompleterMatchFromStart</tabstop>
  <tabstop>mNofColumns</tabstop>
  <tabstop>mEditExpression</tabstop>
  <tabstop>mFilterExpression</tabstop>
 </tabstops>
 <resources>
  <include location="../../../images/images.qrc"/>
 </resources>
 <connections/>
</ui>
